lib/cyclonedx/report_node_modules_cyclonedx.rb

```diff
@@ -6,5 +6,13 @@ class ReportNodeModules < Base
     def initialize(scan_report)
       super(scan_report)
     end
+
+    # Add scope attribute, if available:
+    # https://cyclonedx.org/docs/1.3/json/#components_items_scope
+    def build_component(dependency)
+      super.merge({
+                    "scope": dependency[:scope] || "required"
```

Comment:
Minor nit, "required" is here and down in the reporting modules - can we place it in a const maybe over in the CycloneDX class and then reference that here and in the reporting modules?
SCOPE_REQUIRED = "required".freeze
SCOPE_OPTIONAL =  "optional".freeze